src/subdataframe/subdataframe.jl

```diff
@@ -136,13 +136,16 @@ end
 
 Base.copy(sdf::SubDataFrame) = parent(sdf)[rows(sdf), parentcols(index(sdf), :)]
 
-function without(df::SubDataFrame, icols::Vector{<:Integer})
-    newcols = setdiff(1:ncol(df), icols)
-    view(df, newcols)
-end
 deleterows!(df::SubDataFrame, ind) =
     throw(ArgumentError("SubDataFrame does not support deleting rows"))
 
-DataFrame(sdf::SubDataFrame) = sdf[:, :]
+function DataFrame(sdf::SubDataFrame, copycolumns::Bool=true)
+    if copycolumns
+        sdf[:, :]
+    else
+        DataFrame(eachcol(sdf, false), names(sdf), copycolumns=false)
```

Comment:
Do we want to allow for this (this will create a `DataFrame` containing views of vectors)

Reply:
Why not, that seems consistent. But `eachcol(sdf)` is enough now.

Reply:
I consistently use `eachcol(df, false)` everywhere even if this is not strictly needed to be more explicit (I can change it if we decide to do the same in https://github.com/JuliaData/DataFrames.jl/pull/1752 and rebase).

Reply:
I'd rather use the short form everywhere. It's only more explicit because you're used to `eachcol` having the old behavior. Imagine what would happen if we specified all optional arguments all the time. :-)

Reply:
OK - I will fix it when https://github.com/JuliaData/DataFrames.jl/pull/1752 is merged and this PR rebased.